index.js

```diff
@@ -20,107 +20,108 @@ function error(...msg) {
   console.log('ERR::' + prog, ...msg);
 }
 
-function scrapeDescription(x) {
-  var ret = (x.type === 'text/plain') ? x.content : undefined;
-  return ret;
-}
-
-function convertToJson(body) {
-  // wrapped conversion, not every request/response contains well formed json
-  try {
-    return JSON.parse(body)
-  }
-  catch (e) {
-    return body ? body: '';
-  }
-}
+function formatTimestamp(timestamp) {
+  const date = new Date(timestamp);
+  const offset = date.getTimezoneOffset() * -1;
+  const offsetHours = Math.floor(Math.abs(offset) / 60).toString().padStart(2, '0');
+  const offsetMinutes = (Math.abs(offset) % 60).toString().padStart(2, '0');
+  const offsetSign = offset >= 0 ? '-' : '+';
 
-function compactHeaders(allHeaders) {
-  var h = {};
-  allHeaders.members.forEach(function (x) {
-    h[x.key] = x.value;
-  });
-  return h;
+  return `${date.toISOString().slice(0, -1)}${offsetSign}${offsetHours}:${offsetMinutes}`;
 }
 
 function createLightSummary(rawDetail, options) {
-  let detail = {};
-  Object.assign(detail, {
-    'name': rawDetail.collection.name,
-    'description': scrapeDescription(rawDetail.collection.description),
-    'env': rawDetail.environment.name,
-    'duration': rawDetail.run.timings.completed - rawDetail.run.timings.started,
-    'started': rawDetail.run.timings.started,
-  });
+  if (options.xrayJsonProjectKey == undefined){
```

Comment:
leave this function as it is. Create another function called createXraySummary and put all your code in that.